test/integration/pipeline/tx_pipeline_integration_test_fixture.hpp

```diff
@@ -93,65 +93,32 @@ class TxPipelineIntegrationTestFixture
   }
 
   /**
-   * @param transactions
-   * @param expected_block_height - change expected height when using additional
-   * initial blocks
+   * @param transactions in order
    */
-  void sendTransactions(std::vector<iroha::model::Transaction> transactions) {
-    // generate expected proposal
-    expected_proposal = std::make_shared<iroha::model::Proposal>(transactions);
-    expected_proposal->height = 2;
-
-    // generate expected block
-    expected_block = iroha::model::Block{};
-    expected_block.height = expected_proposal->height;
-    expected_block.prev_hash = genesis_block.hash;
-    expected_block.transactions = transactions;
-    expected_block.txs_number = transactions.size();
-    expected_block.created_ts = 0;
-    expected_block.merkle_root.fill(0);
-    expected_block.hash = iroha::hash(expected_block);
-    irohad->getCryptoProvider()->sign(expected_block);
-
-    // send transactions to torii
-    for (const auto &tx : transactions) {
-      auto pb_tx =
-          iroha::model::converters::PbTransactionFactory().serialize(tx);
-
-      google::protobuf::Empty response;
-      irohad->getCommandService()->ToriiAsync(pb_tx, response);
-    }
-  }
-
-  void validate() {
-    // verify proposal
-    auto proposal_wrapper = make_test_subscriber<CallExact>(
-        irohad->getPeerCommunicationService()->on_proposal(), 1);
-    proposal_wrapper.subscribe(
-        [this](auto proposal) { proposals.push_back(proposal); });
-
-    // verify commit and block
-    auto commit_wrapper = make_test_subscriber<CallExact>(
-        irohad->getPeerCommunicationService()->on_commit(), 1);
-    commit_wrapper.subscribe([this](auto commit) {
-      auto block_wrapper = make_test_subscriber<CallExact>(commit, 1);
-      block_wrapper.subscribe([this](auto block) { blocks.push_back(block); });
-    });
-    irohad->getPeerCommunicationService()->on_commit().subscribe(
-        [this](auto) { cv.notify_one(); });
-
-    // wait for commit
-    std::unique_lock<std::mutex> lk(m);
-
-    cv.wait_for(lk, 10s, [this] { return blocks.size() == 1; });
-
-    ASSERT_TRUE(proposal_wrapper.validate());
-    ASSERT_EQ(1, proposals.size());
-    ASSERT_EQ(*expected_proposal, proposals.front());
-
-    ASSERT_TRUE(commit_wrapper.validate());
-    ASSERT_EQ(1, blocks.size());
-    ASSERT_EQ(expected_block, blocks.front());
+  void sendTxsInOrderAndValidate(
+      const std::vector<iroha::model::Transaction> &transactions) {
+    // test subscribers can't solve duplicate func call.
+    ASSERT_FALSE(duplicate_sent);
```

Comment:
`ASSERT_FALSE(duplicate_sent.fetch_or(true));` or smth like that

Reply:
Thanks. If lack of this, atomic_bool is useless.